cdap-ui/app/cdap/components/Experiments/CreateView/SplitDataStep/index.js

```diff
@@ -37,94 +38,141 @@ const getSplitLogsUrl = (experimentId, splitInfo) => {
   return `${hostPort}${baseUrl}${queryParams}`;
 };
 
-const renderSplitBtn = (experimentId, splitInfo, onClick) => {
-  let isSplitCreated = Object.keys(splitInfo).length;
-  let splitStatus = (splitInfo || {}).status;
-  let isSplitComplete = ['Complete', 'Failed'].indexOf(splitStatus) !== -1;
-  let isSplitFailed = splitStatus === 'Failed';
-  const splitError = () => {
+const getSplitFailedElem = (experimentId, splitInfo) => {
+  return (
+    <span className="split-error-container">
+      Failed to split: Please check {" "}
+      <a href={getSplitLogsUrl(experimentId, splitInfo)} target="_blank"> Logs </a>{" "}
+      for more information
+    </span>
+  );
+};
+
+class SplitDataStep extends Component {
+  state = {
+    shouldShowSplitFailedAlert: false
+  };
+
+  static propTypes = {
+    splitInfo: PropTypes.object,
+    schema: PropTypes.object,
+    createSplitAndUpdateStatus: PropTypes.func,
+    setSplitFinalized: PropTypes.func,
+    setModelCreateError: PropTypes.func,
+    experimentId: PropTypes.string,
+    error: PropTypes.string
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.splitInfo.status && ['CREATING', 'Splitting'].indexOf(this.props.splitInfo.status) !== -1 && nextProps.splitInfo.status === 'Failed') {
```

Comment:
Try splitting this condition into variables and check those variables in the `if` condition. It makes it easier to read the condition and understand when we show split failed alert.

```
const isValidSplitStatus = this.props.splitInfo.status;
const wasSplitting = ['CREATING', 'Splitting'].indexOf(this.props.splitInfo.status);
const isCurrentSplitStatusFailed = nextProps.splitInfo.status === 'Failed';

if (isValidSplitStatus && wasSplitting && isCurrentSplitStatusFailed) {
  ...
}
```
something like that.